Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -938,20 +939,32 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
       info verbosity (show (ghcOptPackages ghcSharedLinkArgs))
 
       whenVanillaLib False $ do
+        -- cbits archive
+        let cLikeObjsPath = map (libTargetDir </>) cLikeObjs
+        unless (null cLikeObjs) $ do
+          Ar.createArLibArchive verbosity lbi (cbitsLibFilePath -<.> (objExtension ++ "_cbits.a")) cLikeObjsPath
         Ar.createArLibArchive verbosity lbi vanillaLibFilePath staticObjectFiles
```

Comment:
So it seems that the regular `libHSxxx.a` archives still contain the cbits objects, and the cbits objects are additionally copied twice into the `libHSxxx_cbits.a`. For users that don't use ghc-wpc yet, this means a small but non-zero extra space overhead, would be nice to have a flag to make it optional.

Reply:
Why are cbits objects copied twice into the `libHSxxx_cbits.a` file? Can you please elaborate?

Reply:
See https://github.com/grin-compiler/cabal/blob/fec6e98860244329ba74a0e4b7e13e4793f4bf14/Cabal/src/Distribution/Simple/GHC.hs#L851. The original `libHSxxx.a` archive contain `staticObjectFiles` which include the cbits objects, and now the `libHSxxx_cbits.a` will include them again.

Reply:
It is true that the regular `libHSxxx.a` archives also contain the cbits objects but they are mixed with Haskell objects. I think this origin information is lost at this point, so it will be impossible or hard to extract out the cbit objects from the `libHSxxx.a` archive. And the dynamic cbit object files are not archived at all, only in cbits archive.

Reply:
At first attempt I wanted to add an extra cabal flag, but I changed my mind later for UX simplicity reasons. The overhead is almost zero in terms of nowadays storage size standards. IMO the UX simplicity win overcomes the minimal size overhead.

Reply:
> this origin information is lost at this point, so it will be impossible or hard to extract out the cbit objects from the libHSxxx.a archive

I don't get it; at this point we're still within the function body of `buildOrReplLib`, all the bindings, including `staticObjectFiles` , `hObjs`, `cLikeObjs`, `stubObjs`, they are all available, and it's clear which objects are haskell and which are non-haskell.

> And the dynamic cbit object files are not archived at all, only in cbits archive.

The dyn cbits objects are not archived indeed, but they are a part of `ghcSharedLinkArgs` and they get linked in the `.so` file. So the space overhead issue is the same with the static case.

Overall, I think while it might be nice to truly ensure the cbits objs only go into cbits archive, that surely requires some extra  burden on you implementation. So copying twice is not really a big issue, but I do believe this should be optional for end users.

Reply:
IMO an additional cabal flag would just make Cabal even more complicated for nothing. What would it gain in terms of UX and maintenance?

Reply:
> > this origin information is lost at this point, so it will be impossible or hard to extract out the cbit objects from the libHSxxx.a archive
> 
> I don't get it; at this point we're still within the function body of `buildOrReplLib`, all the bindings, including `staticObjectFiles` , `hObjs`, `cLikeObjs`, `stubObjs`, they are all available, and it's clear which objects are haskell and which are non-haskell.

I mean that after the archive is built, the hs non-hs obj distinction info is lost for third party tools. (i.e. ext stg interpreter)

Reply:
BTW it is possible to fix the (almost zero) size overhead if the cbits would be left out from libHSxxx.a archive, and the cbits archive would be included in `hs-libraries`. IMO this would work, but I also think that this would be overengineering.

Reply:
My vague sense is that this should be behind a flag, even if said flag is enabled-by-default, the principle reason being  that the flag serves as something to attach documentation to and a means of identifying in the implementation *why* this archive is being produced.